core/src/test/scala/com/payalabs/scalajs/react/bridge/test/GenericComponent.scala

```diff
@@ -0,0 +1,13 @@
+package com.payalabs.scalajs.react.bridge.test
+
+import scala.scalajs.js
+
+import com.payalabs.scalajs.react.bridge.ReactBridgeComponent
+
+
+/**
+  * This is here to "test" (at compile-time) that ReactComponentBridge works with generics
+  */
+object GenericComponent extends ReactBridgeComponent {
+  def apply[A <: js.Any](param: A): Any = auto
```

Comment:
Is there a reason you are declaring the return type of `apply` as `Any` (instead of the recommended `WithProps`). In any case, this code doesn't compile, unless I import `WithProps`.

```
[error] /Users/ramnivas/open-source/scalajs-react-bridge/core/src/test/scala/com/payalabs/scalajs/react/bridge/test/GenericComponent.scala:12: not found: type WithProps
[error]   def apply[A <: js.Any](param: A): Any = auto
[error]                                           ^
[error] one error found
```

Reply:
No specific, I think I changed it around during testing/troubleshooting and forgot to change it back. But with the macro hygiene fixes it doesn't need to be imported and it does compile. When I split it up into separate branches I didn't try compiling it.

For some reason I couldn't get it to work without an explicit return type. It complains about `apply` being recursive which I couldn't understand. I think that's both before and after any/all of my changes.